/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/trl/trainer/ppo_trainer.py:257: UserWarning: No dataset is provided. Make sure to set config.batch_size to the correct value before training.
  warnings.warn(
Loading experiment data Linear Equation
Loading dataset from data/input/LinearEquation.csv
Dataset data/input/LinearEquation.csv not found
Creating dataset for Linear Equation
Dataset saved in data/input/LinearEquation.csv
Current epoch: 0









 95%|████████████████████████████████████████████████████████████████▊   | 122/128 [00:19<00:00,  6.77it/s]
Error: Input contains NaN.
Error: Found input variables with inconsistent numbers of samples: [100, 1]
Error: Found input variables with inconsistent numbers of samples: [100, 1]
Error: Input contains NaN.
Error: Input contains NaN.
Error: Input contains NaN.
Error: Found input variables with inconsistent numbers of samples: [100, 1]
Expression zoo cannot be evaluated. Error: 'ComplexInfinity'
Error: Found input variables with inconsistent numbers of samples: [100, 1]
Error: Found input variables with inconsistent numbers of samples: [100, 1]
100%|████████████████████████████████████████████████████████████████████| 128/128 [00:20<00:00,  6.32it/s]
<lambdifygenerated-3>:2: RuntimeWarning: invalid value encountered in log
  return 4*log(x)
<lambdifygenerated-19>:2: RuntimeWarning: invalid value encountered in sqrt
  return -sqrt(x) + 3*x
<lambdifygenerated-20>:2: RuntimeWarning: invalid value encountered in sqrt
  return sqrt(x) + x + tanh(x)
<lambdifygenerated-26>:2: RuntimeWarning: invalid value encountered in sqrt
  return (sqrt(x) + 3)**(2*x)
You're using a GPT2TokenizerFast tokenizer. Please note that with a fast tokenizer, using the `__call__` method is faster than using a method to encode the text followed by a call to the `pad` method to get a padded encoding.
  0%|                                                                              | 0/128 [00:00<?, ?it/s]
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
Working with expression:  2 * self.x + 3
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
max mean: 0
 max top: 0
Current epoch: 1
Learning rate:  1.5e-05
mean: 0.06268765777349472
top: 0.4354456961154938
invalid: 98









 92%|██████████████████████████████████████████████████████████████▋     | 118/128 [00:18<00:01,  6.66it/s]
Error: Input contains NaN.
Error: Input contains NaN.
Error: Input contains NaN.
Error: Found input variables with inconsistent numbers of samples: [100, 1]
100%|████████████████████████████████████████████████████████████████████| 128/128 [00:19<00:00,  6.64it/s]
<lambdifygenerated-42>:2: RuntimeWarning: invalid value encountered in log
  return log(x)
<lambdifygenerated-52>:2: RuntimeWarning: invalid value encountered in arccos
  return arccos(2*x)
<lambdifygenerated-85>:2: RuntimeWarning: invalid value encountered in log
  return log(5*x)
<lambdifygenerated-87>:2: RuntimeWarning: invalid value encountered in log
  return 2*log(x) + 2*cos(x)
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
Working with expression:  2 * self.x + 3
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
max mean: 0.06268765777349472
 max top: 0.4354456961154938
Current epoch: 2
Learning rate:  1.3500000000000001e-05
mean: 0.10590849816799164
top: 1.0
invalid: 84
Current epoch: 2



 39%|██████████████████████████▉                                          | 50/128 [00:08<00:12,  6.06it/s]
Traceback (most recent call last):
  File "/home/augusto/SR_Methods/symbolic/main.py", line 59, in <module>
    main(params)
  File "/home/augusto/SR_Methods/symbolic/main.py", line 51, in main
    agent.train()
  File "/home/augusto/SR_Methods/symbolic/Agent.py", line 163, in train
    response = ppo_trainer.generate(query.to(self.device), **generation_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/trl/trainer/ppo_trainer.py", line 491, in generate
    response = self.accelerator.unwrap_model(self.model).generate(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/trl/models/modeling_value_head.py", line 202, in generate
    return self.pretrained_model.generate(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/transformers/generation/utils.py", line 1764, in generate
    return self.sample(
           ^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/transformers/generation/utils.py", line 2861, in sample
    outputs = self(
              ^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/transformers/models/gpt2/modeling_gpt2.py", line 1074, in forward
    transformer_outputs = self.transformer(
                          ^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/transformers/models/gpt2/modeling_gpt2.py", line 888, in forward
    outputs = block(
              ^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/transformers/models/gpt2/modeling_gpt2.py", line 390, in forward
    attn_outputs = self.attn(
                   ^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1511, in _wrapped_call_impl
    return self._call_impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/torch/nn/modules/module.py", line 1520, in _call_impl
    return forward_call(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/transformers/models/gpt2/modeling_gpt2.py", line 331, in forward
    attn_output, attn_weights = self._attn(query, key, value, attention_mask, head_mask)
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/augusto/miniconda3/envs/sr/lib/python3.11/site-packages/transformers/models/gpt2/modeling_gpt2.py", line 201, in _attn
    mask_value = torch.full([], mask_value, dtype=attn_weights.dtype).to(attn_weights.device)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
KeyboardInterrupt